### ISSUE TITLE
GET /Checkout でclosed_atを元に取得できる手段がほしい

### ISSUE THREAD

**Issue by mgtae** [ubiregiinc/ubiregi-api#96]
/accounts/account_id/checkouts?since= で会計情報を取得する際
updated_at(更新日)を元にしているとの事だが
updated_atは事前に確認できる術がないので
別途、意図した会計データが取れるよう closed_atを元に取得できる手段がほしい。

**Comment by youchan** [comment]
`closed_at`を元にということですが、実は内部では`business_date`という`closed_at`(もしくは`deleted_at`)の日付の情報をもつようになっています。
`closed_at`を元に取得より`business_date`を使った方が利便性が高いのではと考えています。

`business_date`(つまり日付指定)で取得するということで要望を満すことは可能でしょうか?

**Comment by mgtae** [comment]
すみません。
私の書き方が悪かったと思います。
business_dateだと こちらの意図とは少しずれてしまいます。

下記 2つの使い方をしたいので
「closed_atを元に取得できる手段がほしい」という要望でした。

(1)毎時会計データの自動取り込みをしています
updated_atは 変更される可能性があり、変更された場合に二重取り込みをしてしまうので
変更されない日時を元に取得したい。

(2)意図した会計データを取得したいので
WEB管理画面等で確認できない updated_atではなく closed_atで取得したい。


closed_atではなく 「変更されない日付」であれば 他の項目でもよいのですが。いかがでしょうか。

**Comment by pomu0325** [comment]
(2)の方は日付である必要がありますか？ id, guidなどで取得できればいいのでしょうか？

**Comment by pomu0325** [comment]
> 変更された場合に二重取り込みをしてしまうので変更されない日時を元に取得したい。

`closed_at`ベースですと、逆に会計の取消がされた場合にそれが取得できません。

**Comment by mgtae** [comment]
> (2)の方は日付である必要がありますか？ id, guidなどで取得できればいいのでしょうか？

はい。id, guidが WEB管理画面、もしくはユビレジアプリ上で確認できる方法があれば、  id, guidで取得する事で（2）は解決します。

**Comment by mgtae** [comment]
> closed_atベースですと、逆に会計の取消がされた場合にそれが取得できません。

なるほど。 
ただ、弊社の場合は、基幹システムに取り込んだ後
処理前に 取消されているかどうかのチェックを行っているので 「取り消されたものを再度取得する」 必要はありません。

現状は「同期処理を想定した updated_at を元に範囲指定する方法で取得」 のみなので

「順次会計データを取得する為 closed_at を元に範囲指定する方法で取得」があると助かります。
（※closed_atにこだわる必要はなく、その他変更されない日付であればよいのですが）

**Comment by pomu0325** [comment]
> (1)毎時会計データの自動取り込みをしています
> updated_atは 変更される可能性があり、変更された場合に二重取り込みをしてしまうので
> 変更されない日時を元に取得したい。

こちらは、二重取り込みをしないようにguid, idなどでチェックをしてください。

> (2)意図した会計データを取得したい

どのような用途で「WEB管理画面、もしくはユビレジアプリ上で確認できる」情報から会計データを取得したいのでしょうか？

**Comment by kubibon** [comment]
弊社担当者が変わりましたため、別アカウントから質問させていただきます。

まず、弊社では、ユビレジサーバから会計データを期間指定で基幹システムへ取り込んでおります。
この際、前回取得時の時刻を保持しておき、次回取得時にはその時刻以降の会計データを取り込むようにしております。

ところが、ユビレジサーバ内で会計データの updated_at の値が、過去の日時に勝手に変わってしまう場合があり、
保持しておいた時刻以降の会計データの取り込み対象から外れてしまい、
基幹システムに取り込まれないことが起きています。

このように、ユビレジサーバ内で updated_at の値が勝手に変更されてしまった場合は、
取りこぼしが起きたかどうかも気づかず、月次締めの際のチェックで、
基幹システムに取り込まれていないことに気づき、
どの会計データが取り込まれていないのかを調査することも非常に困難なのです。

最初にご回答いただいている「business_date」で期間指定をして取得できれば、
弊社の要求は満たせると考えております。

ご検討のほどよろしくお願いいたします。

**Comment by belaxbiz** [comment]
ご返信が遅くなりまして申し訳ございません。

恐れ入りますが、昨年末より、ユビレジAPIへの質問回答を内容とする「開発者サポート」を有償化させて頂きました。

もしご購入を検討頂ける場合、 eva@ubiregi.com までご連絡をお願いしても宜しいでしょうか。価格等をご案内させて頂きます。
